docs/ref/models/lookups.txt

```diff
@@ -94,6 +94,19 @@ following methods:
     ``compiler.compile(expression)`` should be used. The ``compiler.compile()``
     method will take care of calling vendor-specific methods of the expression.
 
+    ``kwargs`` may be used to supply extra context to the ``as_sql``-method
+    for that particular call. This can be useful for Expressions in
+    particular, where ``template`` can be overridden and variables intended
+    for interpolating the ``template``-string can be passed. 3rd-party
+    backends may have different SQL-syntax so they can pass these differences
+    through via ``kwargs``.
+
+    Example::
+
+        self.as_sql(compiler, connection,
+            template="%(function)s(%(arg)s)", function='Print', arg='Hello World'
+        )
```

Comment:
(Sorry I took so long to respond, I didn't notice Tim's response until today).

The difference was mostly intentional. Showing `self.coalesce()` here isn't relevant to the example we're highlighting, and is confusing without prior context (ensuring all arguments are non-null). The ConcatPair name is an implementation detail which isn't relevant to the example either.

That said, my example still lacks context by excluding the class definition. Also, if we're going to link to an actual example in code, it's probably important that the code in the docs matches the code we're linking to.

Happy to go with Tim's example here.